src/workerd/api/node/async-hooks.c++

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+#include "async-hooks.h"
+#include <kj/vector.h>
+
+namespace workerd::api::node {
+
+jsg::Ref<AsyncLocalStorage> AsyncLocalStorage::constructor(jsg::Lock& js) {
+  return jsg::alloc<AsyncLocalStorage>();
+}
+
+v8::Local<v8::Value> AsyncLocalStorage::run(
+    jsg::Lock& js,
+    v8::Local<v8::Value> store,
+    v8::Local<v8::Function> callback,
+    jsg::Varargs args) {
+  kj::Vector<v8::Local<v8::Value>> argv(args.size());
+  for (auto arg : args) {
+    argv.add(arg.getHandle(js));
+  }
+
+  js.setAsyncContextTrackingEnabled();
```

Comment:
Can we actually do this in the constructor of (non-root) `AsyncContextFrame`? Putting it here seems like it's fragile -- any other use of `AsyncContextFrame` in the future has to know to enable tracking. If it's just automatic in the constructor of `AsyncContextFrame` it's hard to get wrong in future code.

We could also then remove `setAsyncContextTrackingEnabled()` from the `jsg::Log` interface entirely, since `AsyncContextFrame` can access the internals directly.